orangecontrib/text/pubmed.py

```diff
@@ -0,0 +1,480 @@
+            warnings.warn(
+                'An exception occurred ({0}).'.format(e),
+                RuntimeWarning
+            )
+            if self.error_callback:
+                self.error_callback(e)
+            return
+        search_results = Entrez.read(search_handle)
+        search_handle.close()
+
+        # Store data from this search.
+        self.record_id_list = search_results.get('IdList')
+        self.search_record_count = int(search_results.get('Count'))
+        self.search_record_web_env = search_results.get('WebEnv')
+        self.search_record_query_key = search_results.get('QueryKey')
+
+    def _retrieve_record_batch(self, batch_start, batch_size):
+        """Retrieves a PubMed article record batch.
+
+        Retrieval is based on the info recovered by '_search_for_records()'.
+        The batch size is limited by the 'batch_start' and 'batch_size'
+        parameters. Returns a string containing the article info, if execution
+        was successful and returns None otherwise.
+
+        Args:
+            batch_start (int): Specifies the starting index of this record
+                batch.
+            batch_size (int): Specifies the size of this records batch.
+
+        Returns:
+            list: A list of dictionaries that hold the data for each record.
+        """
+        if None in [self.search_record_web_env, self.search_record_query_key]:
+            raise ValueError(  # Perform a search first!
+                    'No WebEnv or QueryKey data in this PubMed class instance.'
+            )
+
+        fetch_handle = Entrez.efetch(
+                db='pubmed',
+                rettype='medline',
+                retmode='text',
+                retstart=batch_start,
+                retmax=batch_size,
+                webenv=self.search_record_web_env,
+                query_key=self.search_record_query_key
+        )
+
+        data = list(Medline.parse(fetch_handle))
+        fetch_handle.close()
+
+        with shelve.open(self.cache_path) as query_cache:
+            for record in data:
+                id_key = record.get('PMID')
+                if id_key is not None:
+                    query_cache[id_key] = record
+            query_cache.close()
+
+        if self.progress_callback:
+            self.progress_callback()
+
+        return data
+
+    def _retrieve_records(self, num_records,
+                          includes_metadata=PUBMED_TEXT_FIELDS,
+                          use_cache=True):
+        """Retrieves the records queried with '_search_for_records()'.
+
+        If retrieval was successful, generates a corpus with the text fields as
+        meta attributes.
+
+        Args:
+            num_records (int): The number of records we wish to retrieve.
+
+        Returns:
+            `orangecontrib.text.corpus.Corpus`: The retrieved PubMed records
+                as a corpus.
+        """
+        corpus = None
+        batch_size = min(self.DEFAULT_BATCH_SIZE, num_records)
```

Comment:
If we take the `min` then maybe we should rename `DEFAULT_BATCH_SIZE ` to `MAX_BATCH_SIZE `?